### ISSUE TITLE
Failed vagrant action prevents further actions by locking the machine

### ISSUE THREAD

**Issue by appsol** [hashicorp/vagrant#3919]
Host: Ubuntu 12.04 x64
Guest: Centos 6.5 x64
Vagrant Version 1.6.2

Similar to #3741 

Vagrant up command fails inconsistently at Booting VM with:

> The SSH connection was unexpectedly closed by the remote end. This
> usually indicates that SSH within the guest machine was unable to
> properly start up. Please boot the VM in GUI mode to check whether
> it is booting properly.

An attempt to then run vagrant reload will result in:

```
INFO batch_action: Enabling parallelization by default.
 INFO batch_action: Disabling parallelization because provider doesn't support it: virtualbox
 INFO batch_action: Batch action will parallelize: false
 INFO batch_action: Starting action: #<Vagrant::Machine:0x00000003aed758> up {:destroy_on_error=>true, :parallel=>true, :provision_ignore_sentinel=>false, :provision_types=>nil}
 INFO machine: Calling action: up on provider VirtualBox (37ee2d6c-3883-40d7-a1aa-62d3319d7c7f)
 INFO environment: Acquired process lock: dotlock
 INFO environment: Released process lock: dotlock
 WARN environment: Process-lock in use: machine-action-b6e253cfd50dfb7274e5b853a2eac27c
 INFO environment: Running hook: environment_unload
 INFO runner: Preparing hooks for middleware sequence...
 INFO runner: 2 hooks defined.
 INFO runner: Running action: #<Vagrant::Action::Builder:0x00000002a4d600>
ERROR vagrant: Vagrant experienced an error! Details:
ERROR vagrant: #<Vagrant::Errors::MachineActionLockedError: An action 'up' was attempted on the machine 'default',
but another process is already executing an action on the machine.
Vagrant locks each machine for access by only one process at a time.
Please wait until the other Vagrant process finishes modifying this
machine, then try again.

If you believe this message is in error, please check the process
listing for any "ruby" or "vagrant" processes and kill them. Then
try again.>

ERROR vagrant: An action 'up' was attempted on the machine 'default',
but another process is already executing an action on the machine.
Vagrant locks each machine for access by only one process at a time.
Please wait until the other Vagrant process finishes modifying this
machine, then try again.

If you believe this message is in error, please check the process
listing for any "ruby" or "vagrant" processes and kill them. Then
try again.
ERROR vagrant: /opt/vagrant/embedded/gems/gems/vagrant-1.6.2/lib/vagrant/machine.rb:176:in `rescue in action'
/opt/vagrant/embedded/gems/gems/vagrant-1.6.2/lib/vagrant/machine.rb:149:in `action'
/opt/vagrant/embedded/gems/gems/vagrant-1.6.2/lib/vagrant/batch_action.rb:82:in `block (2 levels) in run'
```

Stopping the ruby process will then allow the vagrant reload call to proceed.

Thanks

**Comment by vkarry** [comment]
Hi,
I ran into similar issue when trying to run "vagrant up --provider=lxc"
I got it fixed simply by checking to see if any ruby/vagrant process was already running and kill it.
FYI, In my case, the "vagrant up" process initiated by me prior to the current run was still running.
$ ps -ef | grep ruby
$ ps -ef | grep vagrant

$ kill -9 <pid>

$ Restart "vagrant up --provider=lxc"

Hope this helps someone with similar issue in the future.
